Data/Storable/Endian.hs

```diff
@@ -11,31 +13,40 @@
 -- Storable instances with endianness.
 --
 
-module Data.Storable.Endian 
+module Data.Storable.Endian
   (
     LittleEndian(..), BigEndian(..),
     HasLittleEndian(..), HasBigEndian(..)
-  ) 
+  )
   where
 
 import System.ByteOrder
 
-import Foreign.Storable
 import Foreign.Ptr
+import Foreign.Storable
 
 import Unsafe.Coerce
 
-import Data.Word
 import Data.Bits
+import Data.Word
 
 #if defined(__GLASGOW_HASKELL__) && !defined(__HADDOCK__)
 import GHC.Base
-import GHC.Word
 import GHC.Int
+import GHC.Word
 #endif
 
+import GHC.Generics (Generic(..))
+
 newtype LittleEndian a = LE a
+  deriving ( Eq, Ord, Show, Read, Enum
+           , Num, Integral, Fractional, Floating, Real, RealFrac, RealFloat
```

Comment:
In terms of what I actually need/want, it's just `Eq`, `Show` and `Arbitrary` for testing, and `Generic` for code generation (more specifically, so that types containing `LittleEndian` or `BigEndian` can use `DeriveGeneric`).

However, I added in all the numeric classes from thinking about the situation where someone wants to read in data making sure of its endianness, and then rather than needing to unwrap it before applying numeric calculations (as they may wish to write it back out again with the same endianness).  I'm not too fussed about those though.

`Functor` (with `DeriveFunctor`) might be appropriate though.